test/replication-luatest/gh_6126_auto_rejoin_test.lua

```diff
@@ -0,0 +1,51 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+local fio = require('fio')
+
+local g = t.group('gh_6126')
+
+g.before_each(function(cg)
+    cg.cluster = cluster:new({})
+
+    local box_cfg = {
+        replication = server.build_instance_uri('master'),
+    }
+    cg.replica = cg.cluster:build_and_add_server(
+        {alias = 'replica', box_cfg = box_cfg})
+
+    cg.master = cg.cluster:build_and_add_server(
+        {alias = 'master', box_cfg = box_cfg})
+
+    cg.master:start()
+end)
+
+g.after_each(function(cg)
+    cg.cluster:drop()
+end)
+
+g.test_auto_rejoin = function(cg)
+    -- Test that joining replica tolerates ER_READONLY errors from master
+    cg.master:exec(function() box.cfg{read_only = true} end)
+    cg.replica:start({wait_for_readiness = false})
+    local logfile = fio.pathjoin(cg.replica.workdir, 'replica.log')
+    t.helpers.retrying({}, function()
+        t.assert(cg.replica:grep_log('ER_READONLY', nil,
+            {filename = logfile}), 'Can\'t modify data on a read-only'..
+            'instance - box.cfg.read_only is true')
+
+        t.assert(g.master:exec(function()
+            return box.space._cluster:count() == 1
+        end),  'No join while master is read-only')
```

Comment:
If you do that inside of `retrying`, then if it becomes false, you won't notice until the test times out. I would suggest to move this check after `retrying`.

Reply:
Done